bookmarks/views.py

```diff
@@ -1,3 +1,81 @@
+from users.models       import User
+from bookmarks.models   import BookMark 
+from users.utils        import ConfirmLogin
+
+class BookmarkView(View):
+    @ConfirmLogin
+    def post(self, request, home_pk):
+        try:
+            user = User.objects.get(id=request.user)
+            home = Home.objects.get(id=home_pk)
+            if not home_pk in user.bookmarks.values_list("id",flat=True):
+                user.bookmarks.add(home)
+                return JsonResponse({"MESSAGE":"SUCCESS"},status=201)
+            else:
+                return JsonResponse({"MESSAGE":"ALREADY_BOOKED"},status=400)
+        except User.DoesNotExist:
+            return JsonResponse({"MESSAGE":"USER_NOT_FOUND"},status=404)
+        except Home.DoesNotExist:
+            return JsonResponse({"MESSAGE":"HOME_NOT_FOUND"},status=404)
+
+    @ConfirmLogin
+    def delete(self, request, home_pk):
+        try:
+            home = Home.objects.get(id=home_pk)
+            bookmark = BookMark.objects.get(home=home,user_id=request.user).delete()
+            return JsonResponse({"MESSAGE":"DELETE_SUCCESSFULLY"},status=200)
+        except User.DoesNotExist:
+            return JsonResponse({"MESSAGE":"USER_NOT_FOUND"},status=404)
+        except BookMark.DoesNotExist:
+            return JsonResponse({"MESSAGE":"BOOKMARK_NOT_FOUND"},status=404)
+        except Home.DoesNotExist:
+            return JsonResponse({"MESSAGE":"HOME_NOT_FOUND"},status=404)
+
+class BookmarkListView(View):
+    @ConfirmLogin
+    def get(self, request):
+        offset = int(request.GET.get("offset",0))
+        limit  = int(request.GET.get("limit",15))
+        bookmarks = Home.objects.filter(bookmark__user=request.user).prefetch_related(
+                "homeopiton_set",
+                "homeopiton_set__option",
+                "images",
+                "review_set",
+                "homeprice_set",
+                "homeprice_set__price_category"
+            ).select_related("home_type").all()
+        
+        bookmark_list   = {
+            "bookmarks_count" : bookmarks.count(),
+            "bookmarks" : [{
+            "id"                : home.id,
+            "name"              : home.name,
+            "address"           : home.address,
+            "home_type"         : home.home_type.name,
+            "latitude"          : home.latitude,
+            "longitude"         : home.longitude,
+            "home_images"       : [image.url for image in home.images.all()],
+            "home_options"       : [{
+                    "option_name"  : option.option.name,
+                    "option_count" : option.count
+                } for option in home.homeopiton_set.all()],
+            "price_of_1day" : [
+                {
+                    "category" : home_price.price_category.name,
+                    "cost"     : home_price.cost
+                    }for home_price in home.homeprice_set.all() if home_price.price_category.name=="1박비용"],
```

Comment:
```suggestion
                    }for home_price in home.homeprice_set.all() if home_price.price_category.name=="1박비용"],
```
1박비용이라는 가격 카테고리 일때만 이 내용이 나오게 되나요? 
이 조건을 애초에 컨트롤 할 수는 없나요?

Reply:
에어비앤비 북마크 리스트에 있는 숙소 가격은 1박 기준의 가격으로 노출되어있습니다.

그래서 if 문을 사용하여 1박 비용인 경우의 가격을 추출한 상황입니다..

Reply:
그럼 이렇게 하드코딩보다는 코드 상단에 상수로 선언하고, 상수를 할당해주세요.
```
    PRICE_CATEGORY_NAME = '1박비용'
```

Reply:
리뷰주신 부분 수정 반영 하도록 하겠습니다!